### PR TITLE
Build DALI only for major arch versions

### PR DESCRIPTION
**Pull request by JanuszL**
- removes build of device code for minor architecture variants
  leaving only major ones - like keep sm_70 and remove sm_75.
  The device code should be binary compatible without JITing and
  it should reduce the output binary size

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- removes build of device code for minor architecture variants
  leaving only major ones - like keep sm_70 and remove sm_75.
  The device code should be binary compatible without JITing and
  it should reduce the output binary size

### Affected modules and functionalities:
- Cuda CMake

### Key points relevant for the review:
- NA

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2601

### CODE REVIEW

File: cmake/CUDA_utils.cmake [hunk 1]
```diff
@@ -43,9 +43,15 @@ if (${ARCH} MATCHES "aarch64-")
   set(CUDA_known_archs "53" "62" "72" "75")
 elseif (${ARCH} MATCHES "aarch64")
   # aarch64 SBSA, only >=Volta
-  set(CUDA_known_archs "70" "75" "80" "86")
+  # from the whole list/; "70" "75" "80" "86"
+  # we pick only major arch as minor should be compatible without JITing, it should
+  # srhing the output binary
```
Comment (jantonguirao) [inline cmake/CUDA_utils.cmake:48]:
```suggestion
  # shrink the output binary
```

File: cmake/CUDA_utils.cmake [hunk 2]
```diff
@@ -43,9 +43,15 @@ if (${ARCH} MATCHES "aarch64-")
...
+  set(CUDA_known_archs "70" "80")
 else()
-  set(CUDA_known_archs "35" "50" "52" "60" "61" "70" "75" "80" "86")
+  # from the whole list: "35" "50" "52" "60" "61" "70" "75" "80" "86"
+  # we pick only major arch as minor should be compatible without JITing, it should
+  # srhing the output binary
```
Comment (jantonguirao) [inline cmake/CUDA_utils.cmake:53]:
```suggestion
  # shrink the output binary
```

**Review by jantonguirao — APPROVED** [review]

File: cmake/CUDA_utils.cmake (hunk 2 above)
Comment (JanuszL) [reply cmake/CUDA_utils.cmake]: Done

File: cmake/CUDA_utils.cmake (hunk 1 above)
Comment (JanuszL) [reply cmake/CUDA_utils.cmake]: Done

**Comment by JanuszL** [comment]
It shrinks CUDA 11.x wheel from 700MB->450MB.

**Review by klecki — APPROVED** [review]

**Pull request merged**